core/arch/arm/mm/core_mmu_v7.c

```diff
@@ -671,8 +671,10 @@ static void map_memarea(struct tee_mmap_region *mm, uint32_t *ttb)
 	 * TODO: support mapping devices at a virtual address which isn't
 	 * the same as the physical address.
 	 */
-	if (mm->va < (NUM_UL1_ENTRIES * SECTION_SIZE))
-		panic("va conflicts with user ta address");
+	if (ttb == (void *)core_mmu_get_main_ttb_va()) {
```

Comment:
not only mapping a 4kB page. I need to map device memory, iram memory, gic memory in the IRAM mmu table, they are different addresses that may spread all over the 4GB physical address. Also the virtual address of the mapped memory should be the same as the virtual address in the dram mmu table. That's the point I use section mapping, 16KB memory used. No lpae support in my case. The mapping is done when the first time runs into suspend. I do not find a good place to put the code into a common initializaton code path.